airdrop/project/contracts/distributor-contract/src/main.sw

```diff
@@ -42,23 +46,55 @@ impl AirdropDistributor for Contract {
     #[storage(read, write)]
     fn claim(amount: u64, key: u64, proof: Vec<b256>, to: Identity) {
         // The claiming period must be open
-        require(storage.end_block.read() > height(), StateError::ClaimPeriodNotActive);
+        require(
+            storage
+                .end_block
+                .read() > height(),
+            StateError::ClaimPeriodNotActive,
+        );
 
         // Users cannot claim twice
         let sender = msg_sender().unwrap();
-        require(storage.claims.get(sender).try_read().unwrap_or(ClaimState::Unclaimed) == ClaimState::Unclaimed, AccessError::UserAlreadyClaimed);
+        require(
+            storage
+                .claims
+                .get(sender)
+                .try_read()
+                .unwrap_or(ClaimState::Unclaimed) == ClaimState::Unclaimed,
+            AccessError::UserAlreadyClaimed,
+        );
 
         // There must be enough tokens left in the contract
         let asset = storage.asset.read().unwrap();
         require(this_balance(asset) >= amount, AccessError::NotEnoughTokens);
 
         // Verify the merkle proof against the user and amount
-        require(verify_proof(key, leaf_digest(sha256((sender, amount))), storage.merkle_root.read().unwrap(), storage.number_of_leaves.read(), proof), VerificationError::MerkleProofFailed);
+        let param = (sender, amount);
+        let mut result_buffer: b256 = b256::min();
+        let leaf = asm(hash: result_buffer, ptr: param, bytes: size_of_val(param)) {
```

Comment:
Is the `sha256` somehow significantly more expensive or is there another reason for introducing `asm` at the application level?

Reply:
v0.43 introduced changes in how the hashing is performed and so in order to for merkle proof to work it needs to hash as it did previously. This will be updated once this issue is resolved https://github.com/FuelLabs/sway-libs/issues/186